spring-cloud-sleuth-zipkin/src/main/java/org/springframework/cloud/sleuth/zipkin2/sender/RestTemplateSender.java

```diff
@@ -77,6 +81,16 @@ else if (this.encoding == Encoding.JSON) {
 		this.messageEncoder = BytesMessageEncoder.forEncoding(this.encoding);
 	}
 
+	private String buildUrlWithCustomPathIfNecessary(final String baseUrl,
+			final String customApiPath, final String defaultUrl) {
+		if (Objects.nonNull(customApiPath)) {
+			return baseUrl + (baseUrl.endsWith("/") || customApiPath.startsWith("/")
+					|| customApiPath.isEmpty() ? "" : "/") + customApiPath;
```

Comment:
Wouldn't the `/` handling logic be simpler by doing:
```suggestion
			return Path.of(baseUrl, customApiPath).toString();
```

Reply:
Will have a look.

Reply:
@jonatan-ivanov ,
Can we maintain the previous implementation. Using Path.of cause 2 issues:
1. It will be OS dependent. On Windows, it will be an exception if the customApiPath is blank (illegal character). We need blank in order to cater case for new relic at least.
2. Path.of will strip double forward slash to single. So http://something becomes http:/.